torch/fx/experimental/symbolic_shapes.py

```diff
@@ -402,11 +402,8 @@ def sym_and(self, other):  # noqa: F811
     def is_non_overlapping_and_dense(self, sizes, strides):
         return self.is_non_overlapping_and_dense_indicator(sizes, strides).eq(to_node(self, 1))  # type: ignore[attr-defined]
 
-    # Today we error on calling int on a symbolic shape, as this is a very accessible footgun.
     def int_(self):
```

Comment:
Ooh, would implementing `__int__` someday be in the cards? Or is there a really good reason we don't ever want that.

Reply:
This PR also turns on `__int__` on SymInt. SymNode doesn't have any magic methods on it though.

Reply:
Would it make sense for SymInt to subclass `numbers.Integral` then?

Reply:
Yeah, we can do that